matscipy/dftbp_socketcalc.py

```diff
@@ -0,0 +1,220 @@
+# ======================================================================
+# matscipy - Python materials science tools
+# https://github.com/libAtoms/matscipy
+#
+# Copyright (2014) James Kermode, King's College London
+#                  Lars Pastewka, Karlsruhe Institute of Technology
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+# ======================================================================
+
+"""The dftbsocket module.
+
+This python module provides an ASE calculator for DFTB+.
+It operates DFTB+ through the i-PI interface protocol, which is implemented in DFTB+.
+i-PI can be obtained from https://github.com/i-pi/i-pi/tree/master.
+The i-PI dependency is not (yet) compatible with python 3,
+so the DFTB+ wrapper will not work with python 3 until i-PI is fixed.
+
+A usage example is found in the example directory.
+Please mind that this implementation is NOT WELL TESTED!!!
+
+Make sure to use the same socket specification in the ASE calculator and in the dftb+ input file!
+Unix sockets and network sockets are supported (only unix sockets tested so far).
+Please also read the Socket Driver section in the DFTB+ user manual.
+
+Known issues:
+-Stress tensor is not tested. Sign and/or unit could be wrong
+-Not much testing was done, except to see if it runs without errors
+-No parallel tests were done
+-i-PI officialy only allows for triangular cell matrices, but it seems to not have a hard-coded limit.
+ The full cell seems to be passed through the i-PI interface.
+-There could be a transpose-error in the cell, I am not sure what the definition of the i-PI cell is
+-Network sockets not tested
+
+
+Author: Alexander Held
+"""
+
+from __future__ import (
+    division,
+    absolute_import,
+    print_function,
+    unicode_literals
+)
+from ase.calculators.calculator import Calculator, all_changes
+from ipi.engine.forcefields import FFSocket
```

Comment:
I see, this actually uses iPI. Why is this necessary? I think it is not a good idea to have a dependency on a third part package such as iPI in here.

@jameskermode What is your opinion? I think this should be properly ported to be matscipy only.